include/oneapi/dpl/internal/random_impl/exponential_distribution.h

```diff
@@ -0,0 +1,183 @@
+// -*- C++ -*-
+//===-- exponential_distribution.h ---------------------------------------===//
+//
+// Copyright (C) Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// This file incorporates work covered by the following copyright and permission
+// notice:
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+//
+//===----------------------------------------------------------------------===//
+//
+// Abstract:
+//
+// Public header file provides implementation for Exponential Distribution
+
+#ifndef DPSTD_EXPONENTIAL_DISTRIBUTION
+#define DPSTD_EXPONENTIAL_DISTRIBUTION
+
+namespace oneapi
+{
+namespace dpl
+{
+template <class _RealType = double>
+class exponential_distribution
+{
+  public:
+    // Distribution types
+    using result_type = _RealType;
+    using scalar_type = internal::element_type_t<_RealType>;
+    using param_type = scalar_type;
+
+    // Constructors
+    explicit exponential_distribution(scalar_type __lambda = static_cast<scalar_type>(1.0)) : lambda_(__lambda) {}

```

Comment:
I'll make these constructors.
exponential_distribution() : exponential_distribution(static_cast<scalar_type>(1.0)) {}
explicit exponential_distribution(scalar_type __lambda) : lambda_(__lambda) {}

But then if we add following constructor
explicit exponential_distribution(const param_type& parm);
There will be two constructors with equal signature. Because of "using param_type = scalar_type"